docs/guides.md

```diff
@@ -544,67 +544,176 @@ secureboot-certs
+1. Shutdown the VM if it is not already shut down.
 
-Next, setup the VM secure boot state:
+2. From the XCP-ng CLI, invoke the command:
 
 ```
 varstore-sb-state <vm-uuid> setup
 ```
 
-Now, you may [Enable Secure Boot](#enable-secure-boot).
+:::warning
+`varstore-sb-state <vm-uuid> setup` wipes previously installed Secure Boot certificates (if there were any). Upon boot, they will be replaced by the Microsoft certificates installed by the `secureboot-certs` script.
+:::
+
+##### Enabling Secure Boot for an Existing UEFI VM in XO
+
+1. Shutdown the VM if it is not already shutdown.
+
+2. Go to the *Advanced* tab of the VM and click the **Secure boot** toggle
+to enable Secure Boot.
+
+![](../assets/img/screenshots/xo_uefi_sb_post_install_option.png)
 
-#### Enable Secure Boot
+##### Enabling Secure Boot for an Existing UEFI VM in XCP-ng Center
+
+1. Shutdown the VM using the [shutdown](cli_reference.md#vm-shutdown) command if it is not already shut down.
+
+2. Change the firmware to *UEFI with SecureBoot*.
+
+:::warning
+It is only recommend to interchange the firmware between *UEFI* and *UEFI with SecureBoot*, **not** *BIOS*.
+:::
+
+##### Enabling Secure Boot for an Existing UEFI VM using `xe`
+
+1. Shutdown the VM using the [shutdown](cli_reference.md#vm-shutdown) command if it is not already shut down.
+
+2. In the XCP-ng CLI, set the platform Secure Boot mode to `true`:
 
 ```
-# Enable secure boot for a UEFI VM
+# Enable Secure Boot for the VM
 xe vm-param-set uuid=<vm-uuid> platform:secureboot=true
 ```
 
+#### Setup for Linux VMs
+
+Some Linux distributions may require special packages for Secure Boot to function.  Please follow the distribution's documentation to install any required Secure Boot software (e.g., shim) *before* enabling Secure Boot for the VM in XCP-ng.
+
+
+:::warning
+If the VM has any unsigned kernel modules, they will fail to load after enabling Secure Boot.  Furthermore, the distribution will likely restrict other kernel features that are seen as loop holes in Secure Boot (kexec, /dev/mem, etc...).  Please read the Secure Boot documentation from the distribution.
+:::
+
+#### Setup for Windows VMs
+
+Windows VMs do not require extra installation packages because the Windows Loader and kernel are signed by the keys already installed by the `secureboot-certs` script.  Enabling Secure Boot for the VM in XCP-ng enables Secure Boot in the VM UEFI firmware.
+
+:::warning
+If your VMs have any unsigned drivers, they will fail to load after enabling Secure Boot.
+:::
+
 :::tip
-Optionally, the secure boot mode may be toggled in the
-VM/Advanced tab in Xen Orchestra.
+Unlike Windows, Linux is not signed by Microsoft. Therefore it is required to either install, or use a distribution that comes by default with an intermediary boot loader that *is* signed by Microsoft, if booting Linux with Secure Boot.  This boot loader is typically `shim`.
```

Comment:
Why is this tip in the windows section?

Reply:
Because I felt people would be wondering why they're being told different info for Windows than what they were told for Linux.  It can surely be removed or moved.

Reply:
I think it's interesting information that rather belongs to the Linux part. If it's better to read the Windows doc before, we can switch the Linux and Windows sections. Else I'm sure people will be able to see that windows section is just a few lines below.